Sources/StreamVideo/WebRTC/v2/StateMachine/Stages/WebRTCCoordinator+Joined.swift

```diff
@@ -385,17 +385,20 @@ extension WebRTCCoordinator.StateMachine.Stage {
                 return
             }
             let stateAdapter = coordinator.stateAdapter
+            let sessionId = await stateAdapter.sessionID
 
-            let statsReporter = WebRTCStatsReporter(
-                sessionID: await stateAdapter.sessionID
-            )
-
-            statsReporter.interval = await stateAdapter.statsReporter?.interval ?? 0
-            statsReporter.publisher = await stateAdapter.publisher
-            statsReporter.subscriber = await stateAdapter.subscriber
-            statsReporter.sfuAdapter = await stateAdapter.sfuAdapter
+            if await stateAdapter.statsReporter?.sessionID != sessionId {
+                let statsReporter = WebRTCStatsReporter(
+                    sessionID: await stateAdapter.sessionID
+                )
+                await stateAdapter.set(statsReporter)
+            }
 
-            await stateAdapter.set(statsReporter)
+            let statsReporter = await stateAdapter.statsReporter
+            statsReporter?.interval = await stateAdapter.statsReporter?.interval ?? 0
```

Comment:
if it's a zero, we don't report, right?

Reply:
That's right. This is the method that runs whenever the interval property is changing on the `WebRTCStatsReporter`
```swift
private func scheduleCollection(with interval: TimeInterval) {
        guard interval > 0 else {
            log.warning("Collection interval should be greater than 0.")
            collectionCancellable?.cancel()
            return
        }

        collectionCancellable?.cancel()
        collectionCancellable = Foundation
            .Timer
            .publish(every: interval, on: .main, in: .default)
            .autoconnect()
            .sink { [weak self] _ in self?.collectStats() }

        log.debug("Stats collection is now scheduled with interval:\(interval).")
    }
```